src/aks-preview/azext_aks_preview/_params.py

```diff
@@ -163,6 +163,49 @@ def load_arguments(self, _):
         c.argument('kubernetes_version', completer=get_k8s_upgrades_completion_list)
         c.argument('yes', options_list=['--yes', '-y'], help='Do not prompt for confirmation.', action='store_true')
 
+    with self.argument_context('aks maintenanceconfiguration') as c:
+        c.argument('cluster_name', type=str, help='The cluster name.')
+
+    for scope in ['aks maintenanceconfiguration add', 'aks maintenanceconfiguration update']:
+        with self.argument_context(scope) as c:
+            c.argument('config_name', type=str, options_list=['--config-name'], help='The config name.')
```

Comment:
If I understand correctly, in this command, config is the main resource, the option for `config_name` should be named `--name/-n`, the option for `cluster_name` should be named `--cluster-name`.  Check `az aks nodepool add --help`  for reference.